tests/dialects/test_builtin.py

```diff
@@ -58,6 +60,51 @@ def test_IntegerType_size():
     assert IntegerType(64).size == 8
 
 
+def test_IntegerType_normalize():
+    s8 = IntegerType(8, Signedness.SIGNED)
+    u8 = IntegerType(8, Signedness.UNSIGNED)
+
+    assert i8.normalize_value(-1) == -1
+    assert i8.normalize_value(1) == 1
+    assert i8.normalize_value(255) == -1
+
+    assert s8.normalize_value(-1) == -1
+    assert s8.normalize_value(1) == 1
+    assert s8.normalize_value(255) == 255
```

Comment:
I'm not sure I understand the semantics here, do we not change values that fall outside the range?

Reply:
Yep, and your comment means that I should have documented this better: the idea is that we want to first preserve the behaviour for invalid integers, to raise a VerifyException with the invalid value. If the valid value is ambiguous due to signlessness, only then normalize to the signed range. I'll add this somewhere.
- [ ] add more explanation to doc comment